src/cli.js

```diff
@@ -20,18 +26,22 @@ async function main(filename, resourceName, resource, options) {
     await fs.access(resource, constants.R_OK);
     resourceData = await fs.readFile(resource);
   } catch {
-    console.log("Can't read resource file");
+    logger.error("Can't read resource file");
     process.exit(1);
   }
 
   try {
+    logger.info(
+      "Start injection of " + resourceName + " in " + filename + "..."
+    );
     await inject(filename, resourceName, resourceData, {
       machoSegmentName: options.machoSegmentName,
       overwrite: options.overwrite,
       sentinelFuse: options.sentinelFuse,
     });
+    logger.success("💉 Injection done!");
```

Comment:
Are we sure that this won't get printed if `LIEF_ERR` gets called by LIEF?

Reply:
If we got a `LIEF_ERR`, is inject won't throw it?

Reply:
No, inject doesn't throw when LIEF has an error unfortunately

Reply:
What should the approach be as the resolution of inject function doesn't mean that all happen nicely?

The ideal approach for me is to keep all the log stuff as they are in the `src/cli.js` and make sure that the promises reject if a LIFE_ERR happens. But maybe it's not possible, let me know. (I'll check it anyway)

Another approach could simply be to change the message to: "Injection finished" within log info instead of success. And print the LIFE_ERR from the native side.

WDYT?

Reply:
To be clear, I think there might be some inconsistency in how LIEF reports errors.
If you consider this `LIEF_ERR()`:
https://github.com/nodejs/postject/blob/35343439cac8c488f2596d7c4c1dddfec1fddcae/vendor/lief/src/PE/Parser.cpp#L1011
it would break out of the while loop and return `ok()` although it's a `LIEF_ERR`, not a `LIEF_WARN`.

However, LIEF does return errors for some `LIEF_ERR()`s like in here - https://github.com/nodejs/postject/blob/35343439cac8c488f2596d7c4c1dddfec1fddcae/vendor/lief/src/PE/Parser.cpp#L117-L118, which would certainly reject the promise.

I wonder if we should patch LIEF to prepend something like `ERROR: ` / `WARN: ` for `LIEF_ERR()` and `LIEF_WARN()`, etc. And also follow a similar approach in the logs we produce in this file.

Reply:
> I don't understand how it could help; could you elaborate more?

It would make it more clear to our users whether some log from LIEF is an error / a warning. It isn't conveyed correctly programmatically to the caller in some areas of LIEF as I explained in https://github.com/nodejs/postject/pull/73#discussion_r1115569271, so this should help the end users understand the issue better.

> Also I think as a first step I could just change the message as I proposed:

I'm okay with keeping the success part as long as we have a way to convey whether something is a warning / error to the users of postject.

Reply:
> It would make it more clear to our users whether some log from LIEF is an error / a warning. It isn't conveyed correctly programmatically to the caller in some areas of LIEF as I explained in https://github.com/nodejs/postject/pull/73#discussion_r1115569271, so this should help the end users understand the issue better.

I understand this, but then it will be still weird to have in the console: 

```
WARN: bla bla
ERROR: bla bla
Injection done!
```

Anyway, my other question is should we wait until I raise the PR on LIEF for merging this one?

I think the modification should happen there: https://github.com/lief-project/LIEF/blob/2aad8e8de3a6c63ef5568114ae4e1d43a6cb7b84/src/logging.hpp

Reply:
> but then it will be still weird to have in the console

Ah, are u referring to the `Injection done!` part? Should we remove that part and instead follow the alternate approach Matteo mentioned in the issue - `Another option is to clearly specify what is a warning and what is an error.`?

> Anyway, my other question is should we wait until I raise the PR on LIEF for merging this one?

It doesn't have to be a PR in LIEF though. It can be a patch in postject itself because we vendor the LIEF source code. We could send a PR to LIEF too but that's optional.